### ISSUE TITLE
[4.0] Back-end. Question concerning sidebar removal

### ISSUE THREAD

**Issue by ReLater** [joomla/joomla-cms#24742]
Several sidebars and logic have been removed already in J4 backend. Because I fiddle a bit about custom backend templates **just a question** which code concerning sidebars will be removed in Joomla 4 (I don't want to start a new discussion here).

As example: Removal of com_templates sidebar (https://github.com/joomla/joomla-cms/pull/24092)

Now I would like to know if codes like
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_templates/View/Templates/HtmlView.php#L113
and
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_templates/Helper/TemplatesHelper.php#L34-L46
will also be removed without replacement from all Joomla extensions.

And if the class JHtmlSidebar will be removed, too?

@brianteeman Do you have detailed informations yet?

**Comment by infograf768** [comment]
> And if the class JHtmlSidebar will be removed, too?

I hope not. Some 3rd party components use it.

**Comment by ghost** [comment]
@coolcat-creations should know more.

**Comment by coolcat-creations** [comment]
As far as I understood there is no real sidebar position. If Component creators use it they create their own sidebar thing. The removal of the sidebar in Joomla is because we have a sidemenu and the items go into that one. At least I guess so. Maybe @bembelimen can confirm or clarify.

**Comment by brianteeman** [comment]
Not sure why you tagged me in this

**Comment by bembelimen** [comment]
The Sidebar (and its JHtmlSidebar classes) will not be removed, but the core will not use them anymore (at least tries it). Because with the left menu you'll lose a lot of space, when using the sidebar.

Probably the layout position will change in some way, but the functionality will still exists.

**Comment by ReLater** [comment]
@brianteeman 
> Not sure why you tagged me in this

Because it's your pr but didn't remove the lines that I've linked to in https://github.com/joomla/joomla-cms/issues/24742#issue-437966913 When I look into the codes for articles view `public static function addSubmenu($vName)` is an empty method now. Thus I thought you know more about the next steps.

Others: 

My question is not concerning Atum. I understood the decision is incontrovertible for that template.

I'm talking about **custom** backend templates. When the code lines that I've linked to in https://github.com/joomla/joomla-cms/issues/24742#issue-437966913 remain untouched in core I just have to add a sidebar position in my custom template and the sidebar can be populated easily in overrides.

If lines and methods like https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_templates/Helper/TemplatesHelper.php#L34-L46 are removed or cleaned out completely I have to build my own classes and methods inside the template codes/overrides.

That's all I'm asking ;-) "I hate redundant code in my extensions."
